lib/tasks/trends.rake

```diff
@@ -29,4 +29,102 @@ namespace :trends do
     end
   end
 
+  desc 'Reduces amount of points for each trend group'
+  task :reduce => :environment do
+    start = Time.now
+
+    trends = Trend.pluck(:id)
+    trends_count = trends.length
+    current_record = 0
+
+    trends.each do |trend_id|
+      puts "Working on trend_id #{trend_id}, #{(current_record += 1)} of #{trends_count}"
+
+      current_interval = TrendCounter.where(trend_id: trend_id).order(:created_at).first
```

Comment:
No need, the current version of the task is idempotent - you can run it more than once. It won't ruin the data.